Source/WebKit2/WebProcess/WebPage/wpe/WebPageWPE.cpp

```diff
@@ -54,6 +54,7 @@ void WebPage::platformPreferencesDidChange(const WebPreferencesStore& store)
     m_page->settings().setAllowRunningOfInsecureContent(store.getBoolValueForKey(WebPreferencesKey::allowRunningOfInsecureContentKey()));
     m_page->settings().setAllowDisplayOfInsecureContent(store.getBoolValueForKey(WebPreferencesKey::allowDisplayOfInsecureContentKey()));
     m_page->settings().setScrollToFocusedElementEnabled(store.getBoolValueForKey(WebPreferencesKey::scrollToFocusedElementEnabledKey()));
+    m_page->settings().setShouldBlockDatabaseStorage(!store.getBoolValueForKey(WebPreferencesKey::databasesEnabledKey()));
```

Comment:
All the changes to Document, SecurityOrigin and Settings aren't needed -- I think it should be simpler to add a new RuntimeEnabledFeature entry that in WebPage::updatePreferences() is set to the negative value of WebPreferencesKey::databasesEnabledKey(), and that is then checked in shouldThrowSecurityException() in IDBFactory.cpp where it causes the exception throw if it's set to true.

Note that DatabaseManager::singleton().setIsAvailable() is already called in WebPage::updatePreferences(), where it's set to the value of databasesEnabledKey(). This results in a security exception being thrown in DOMWindowWebDatabase::openDatabase() just the same as if the SecurityOrigin::canAccessDatabase() call returned false there.

Reply:
@zdobersek 
Hi Zan,
don't we need to support separate settings for multiple web pages running in the same web process?
(see WebProcessPool::createNewWebProcessRespectingProcessCountLimit https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/master/Source/WebKit2/UIProcess/WebProcessPool.cpp#L768)

Reply:
In any case to speed up the code review, I've re-raised pull request according to your comments. Please let me know how you think we should proceed.
Thanks!